examples/mpi-mnist/mpi_backend/cpu/Dockerfile

```diff
@@ -0,0 +1,44 @@
+FROM nvidia/cuda:9.0-cudnn7-devel-ubuntu16.04
+ARG PYTHON_VERSION=3.6
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
+         build-essential \
+         cmake \
+         git \
+         curl \
+         vim \
+         ca-certificates \
+         openssh-client \
+         libjpeg-dev \
+         libpng-dev &&\
+     rm -rf /var/lib/apt/lists/*
+
+RUN curl -o ~/miniconda.sh -O  https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh  && \
+     chmod +x ~/miniconda.sh && \
+     ~/miniconda.sh -b -p /opt/conda && \
+     rm ~/miniconda.sh && \
+     /opt/conda/bin/conda install -y python=$PYTHON_VERSION numpy pyyaml scipy ipython mkl mkl-include cython typing && \
+     /opt/conda/bin/conda install -y -c pytorch magma-cuda90 && \
+     /opt/conda/bin/conda install -c conda-forge openmpi && \
+     /opt/conda/bin/conda clean -ya
+ENV PATH /opt/conda/bin:$PATH
+# This must be done before pip so that requirements.txt is available
+WORKDIR /opt/pytorch
+#COPY . .
+RUN git clone --recursive https://github.com/pytorch/pytorch
+RUN cd pytorch/
+RUN ls -lah
```

Comment:
Could we merge these run scripts into one to reduce the size of the docker image?

Reply:
Those are unnecessary. Fixed.